Add Lockbox (DO NOT MERGE TILL AFTER 6/28 deploy and l10n merge)

devinreams opened the pull request:
Fixes https://github.com/mozilla-lockbox/mozilla-lockbox.github.io/issues/47

My bad, I gave bad advice so re-created @nickbrandt's branch here. Replacing #3623

- [x] We want to ensure that this is set to not be live for now.
- [x] We are unsure what ID # should be used for Lockbox in the YAML file
- [x] We need to get all the proper PI/QA folks included
- [x] Some other content may still need adjusted for other areas of the Lockbox Test Pilot instance.
- [x] but make sure it becomes "Featured" when ready to launch
- [x] Final items: https://github.com/mozilla-lockbox/mozilla-lockbox.github.io/issues/44

nickbrandt committed:
Initial commit of setup for new Lockbox experiment, with some assets and team information added.

nickbrandt committed:
Adding avatar and title for Jim

nickbrandt committed:
Adding in copy and images for detail page. WIP

nickbrandt committed:
Updating content

nickbrandt committed:
Updating measurements/privacy section for eventual linking off to Lockbox privacy site.

pdehaan commented on content-src/experiments/lockbox.yaml:
```diff
@@ -0,0 +1,162 @@
+order: 3
+is_featured: true
+platforms: ['mobile']
+web_url: https://example.com/web-experiment
```
Should this be https://lockbox.firefox.com ?

pdehaan commented on content-src/experiments/lockbox.yaml:
```diff
@@ -0,0 +1,162 @@
+gradient_start: '#00BCD8'
+gradient_stop: '#0086DC'
+
+xpi_url: 'https://testpilot.firefox.com/files/experiment-name/latest'
```
Should this be /lockbox/ or /@lockbox/ or something instead of "experiment-name"?

pdehaan replied:
Although, if you're an iOS app without an add-on, you may be able to drop these values.
You may want to check what Send did in https://github.com/mozilla/testpilot/blob/master/content-src/experiments/send.yaml

pdehaan commented on content-src/experiments/lockbox.yaml:
```diff
@@ -0,0 +1,162 @@
+locale_grantlist:
+  - 'de'
+
+dev: false
```
Not sure if we want to merge w/ "dev: true" so we can test on the dev site without it risking riding the trains into production.

pdehaan commented on content-src/experiments/lockbox.yaml:
```diff
@@ -0,0 +1,162 @@
+    title: 'Sr. UX Designer'
+    avatar: /static/images/experiments/lockbox/avatars/nbrandt.png
+  -
+    display_name: 'Peter deHaan'
```
Feel free to remove this 🤡.
I can think of several other QA people far more deserving of credit than this donkey.

devinreams replied:
> can think of several other QA people far more deserving of credit

Agreed. I've added to our checklist above to finalize our contributor list.

pdehaan commented on content-src/experiments/lockbox.yaml:
```diff
@@ -0,0 +1,162 @@
+  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam atque accusamus,
+    suscipit, nam commodi excepturi error modi.</p>
+
+graduation_url: 'http://example.com/graduation-report'
```
I think a lot of the following items are placeholders and may need some :eyes: before merging.
Like, we probably don't want `eol_warning` or `incompatible` here.

pdehaan commented on locales/en-US/experiments.ftl:
```diff
@@ -60,6 +60,29 @@ containersContributors6Title = Front End Security
+firefoxlockboxContributors8Title = Sr. UX Researcher
+firefoxlockboxContributors9Title = Content Strategy Lead
+firefoxlockboxContributors10Title = Sr. UX Designer
+firefoxlockboxContributors11Title = QA Developer
```
If you ✂️me from the `contributors:` list above (and please do), you'll also need to make sure you tweak these values, otherwise all the names+titles will get out of sync (unless you replace me with MBrandt -- although I'm guessing he has a much more glamorous job title than I).

pdehaan commented on content-src/experiments/color.yaml:
```diff
@@ -1,7 +1,7 @@
 title: 'Color'
 slug: color
-is_featured: true
+is_featured: false
```
We may not want to unfeature this yet.
I did make a note about adding `dev: true` to the lockbox.yaml below though, so feel free to ignore this comment temporarily until you read that. At which point you're welcome to ignore this comment permanantly.

devinreams replied:
>We may not want to unfeature this yet.

Good point, it was good to flip for testing but needs to remain in place before merging. 👍

flodolo commented on locales/en-US/experiments.ftl:
```diff
@@ -60,6 +60,29 @@ containersContributors6Title = Front End Security
 containersContributors8Title = Firefox UX
 containersContributors9Title = Firefox UX
+firefoxlockboxDescription = Take your passwords everywhere with Firefox Lockbox, an iOS mobile app.
+firefoxlockboxIntroduction = <p>Firefox Lockbox lets you easily access your passwords everywhere. Logins are stored and synced securely, using 256-bit encryption, keeping your information safe. Easily copy username and password details to get into apps and websites. Never again find yourself locked out of your account while on your phone!</p> <p><strong>Firefox Lockbox requires a Firefox Account  ...
```
HTML markup shouldn't really be in the string. Can this be split up in paragraphs?

Also the actual URL shouldn't be in the localized string.

pdehaan replied:
Sadly, I think we have prior art in excessive HTML markup in introductions in [at least]: min-vid, pageshot, and send:

https://github.com/mozilla/testpilot/blob/a323fac87aeee2591acb24b354906a5fb276d743/locales/en-US/experiments.ftl#L64

https://github.com/mozilla/testpilot/blob/a323fac87aeee2591acb24b354906a5fb276d743/locales/en-US/experiments.ftl#L114

https://github.com/mozilla/testpilot/blob/a323fac87aeee2591acb24b354906a5fb276d743/locales/en-US/experiments.ftl#L130

devinreams replied:
>HTML markup shouldn't really be in the string

Ah yes, this appears to be the result of the l10n extraction. I'd have to find a clever way to redo the YAML so the "introduction" could handle these strings and still be formatted as expected and that's probably beyond my ability and scope of this PR.

flodolo commented on locales/en-US/experiments.ftl:
```diff
@@ -60,6 +60,29 @@ containersContributors6Title = Front End Security
+firefoxlockboxIntroduction = <p>Firefox Lockbox lets you easily access your passwords everywhere. Logins are stored and synced securely, using 256-bit encryption, keeping your information safe. Easily copy username and password details to get into apps and websites. Never again find yourself locked out of your account while on your phone!</p> <p><strong>Firefox Lockbox requires a Firefox Account  ...
+firefoxlockboxDetails0Copy = Access username and password information for all the accounts you save to Firefox.
+firefoxlockboxDetails1Copy = Manage accounts in Firefox and sync changes automatically to Firefox Lockbox.
+firefoxlockboxDetails2Copy = Sort and filter on your accounts to quickly locate the one you need.
```
(non English speaker) Filter "on"?

pdehaan replied:
(poor English speaker) Yeah, I think "on" is unnecessary word here and above in https://github.com/mozilla/testpilot/pull/3624/files#diff-764620a2f7d2d9a3655e4a1806216a95R46

flodolo commented on locales/en-US/experiments.ftl:
```diff
@@ -60,6 +60,29 @@ containersContributors6Title = Front End Security
+firefoxlockboxContributors11Title = QA Developer
+firefoxlockboxContributors12Title = Community Contributor
+firefoxlockboxContributors13Title = Community Contributor
+firefoxlockboxToursteps0Copy = <p>Lorem ipsum dolor sit amet.</p>
```
Uhm… what?

pdehaan replied:
I think this is just a copy of the placeholder text from https://github.com/mozilla/testpilot/pull/3624/files#diff-764620a2f7d2d9a3655e4a1806216a95R110

devinreams committed:
set yaml values to those expected for launch, remove unneeded ones

pdehaan commented on content-src/experiments/lockbox.yaml:
```diff
@@ -0,0 +1,162 @@
+xpi_url: 'https://testpilot.firefox.com/files/experiment-name/latest'
+addon_id: 'experiment-name@mozilla.com'
+
+created: '2016-01-01T00:00:00.000000Z'
```
We also may want something a bit more... future-esque.

devinreams committed:
rearrange for readibility and remove unnecessary stuff

devinreams committed:
remove l10n strings while in dev true mode

devinreams committed:
don't un-feature color experiment

devinreams committed:
better english for filter action and noun

devinreams committed:
add more awesome QA folks and avatars

devinreams committed:
add experiment todos so we don't forget

devinreams committed:
set up as ios experiment with app store url

devinreams committed:
remove placeholder assets, add more todos

pdehaan commented on content-src/experiments/lockbox.yaml:
```diff
@@ -0,0 +1,129 @@
+
+measurements:
+  - >
+    REMOVE ME
```
I haven't looked into TestPilot schemas lately to see if `measurements` is required, but I'll just call this out because it is literally telling me to remove it.

devinreams replied:
Yeah the page dies without something here. Even though it pulls the localized string (not this) into the page itself.

I want to see about making measures _not_ required and not displaying the bulleted list in "Your privacy" section if empty in a follow-up PR.

devinreams committed:
Merge branch 'master' into lockbox2

devinreams committed:
add social sharing icons for facebook and twitter

devinreams committed:
update launch time to 7 AM Pacific

devinreams committed:
add placeholder generic measurement language

devinreams committed:
Merge branch 'master' into lockbox2

devinreams committed:
enable Lockbox experiment and make it featured with a placeholder video

devinreams committed:
Merge branch 'master' into lockbox2

devinreams committed:
replace intro with final copy and links approved by legal

devinreams committed:
whoops, did not save file

devinreams committed:
add missing notice for home page featured section

devinreams committed:
add preamble and remove measurements

anticipating changes from and depends on #3649 #3646
language provided by legal team

devinreams committed:
better placeholder video, still not final

devinreams committed:
Merge branch 'master' into lockbox2

devinreams committed:
remove unused legal notice

not currently available to mobile experiments

devinreams committed:
final video URL

devinreams committed:
Merge branch 'master' into lockbox2

devinreams committed:
Merge branch 'master' into lockbox2
